test/test-lex.cpp

```diff
@@ -136,12 +136,119 @@ TEST(test_lex, lex_binary_numbers) {
   check_single_token(u8"0B010101010101010", token_type::number);
 }
 
+// fails
+TEST(test_lex, DISABLED_fail_lex_binary_number) {
+  {
+    error_collector v;
+    padded_string input(u8"0b1.1");
+    lexer l(&input, &v);
+    EXPECT_EQ(l.peek().type, token_type::number);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::end_of_file);
```

Comment:
Why should `0b1.1` parse as a single number token, but `0o52.2` parse as two number tokens? I think they should be consistent with each other. With your patch, they are consistent; `0b1.1` parses as two number tokens, and `0o52.2` also parses as two number tokens.

Reply:
Agreed